### PR TITLE
SONARKT-340 - Add test framework for KotlinGradleSensor checks

### PR DESCRIPTION
**Pull request by kaufco**

### CODE REVIEW

File: sonar-kotlin-gradle/src/main/java/org/sonarsource/kotlin/gradle/KotlinGradleCheckList.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.kotlin.gradle
+
+import org.sonarsource.kotlin.checks.ProviderGetOutsideTaskCheck
+
+val KOTLIN_GRADLE_CHECKS = listOf(
+    ProviderGetOutsideTaskCheck::class.java,
```
Comment (leveretka) [inline sonar-kotlin-gradle/src/main/java/org/sonarsource/kotlin/gradle/KotlinGradleCheckList.kt:25]: Let's not make this check a part of this PR

File: sonar-kotlin-gradle/src/test/java/org/sonarsource/kotlin/gradle/ProviderGetOutsideTaskCheckTest.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.kotlin.gradle
+
+import org.sonarsource.kotlin.checks.ProviderGetOutsideTaskCheck
+import org.sonarsource.kotlin.testapi.CheckTest
+
+class ProviderGetOutsideTaskCheckTest : CheckTest(ProviderGetOutsideTaskCheck(), isScriptFile = true)
```
Comment (leveretka) [inline sonar-kotlin-gradle/src/test/java/org/sonarsource/kotlin/gradle/ProviderGetOutsideTaskCheckTest.kt:25]: I think we need some dummy check and dummy test for it, not a real one.

File: kotlin-checks-test-sources/build.gradle.kts [hunk 3]
```diff
@@ -19,6 +19,7 @@ dependencies {
     implementation("com.google.guava:guava:30.1.1-jre")
     implementation("org.apache.commons:commons-lang3:3.9")
     implementation("commons-lang:commons-lang:2.6")
+    implementation("org.jetbrains.kotlin:kotlin-script-runtime:1.4.10")
```
Comment (leveretka) [inline kotlin-checks-test-sources/build.gradle.kts:22]: Why such an old version? We use Kotlin 1.8

Comment (leveretka) [reply kotlin-checks-test-sources/build.gradle.kts:22]: Also, if we move these files out from the sources, we don't need this dependency

File: sonar-kotlin-plugin/src/main/resources/org/sonar/l10n/kotlin/rules/kotlin/S6622.html [hunk 4]
```diff
@@ -0,0 +1,48 @@
+<h2>Why is this an issue?</h2>
```
Comment (leveretka) [inline sonar-kotlin-plugin/src/main/resources/org/sonar/l10n/kotlin/rules/kotlin/S6622.html:1]: Remove this file, as this rule implementation shouldn't be a part of this PR

File: sonar-kotlin-plugin/src/main/resources/org/sonar/l10n/kotlin/rules/kotlin/S6622.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (leveretka) [inline sonar-kotlin-plugin/src/main/resources/org/sonar/l10n/kotlin/rules/kotlin/S6622.json:1]: Remove this file, as this rule implementation shouldn't be a part of this PR

File: sonar-kotlin-plugin/src/main/resources/org/sonar/l10n/kotlin/rules/kotlin/Sonar_way_profile.json [hunk 6]
```diff
@@ -112,6 +112,7 @@
     "S6532",
     "S6558",
     "S6611",
-    "S6615"
+    "S6615",
+    "S6622"
```
Comment (leveretka) [inline sonar-kotlin-plugin/src/main/resources/org/sonar/l10n/kotlin/rules/kotlin/Sonar_way_profile.json:116]: Remove this rule from the SonarWay, as this rule implementation shouldn't be a part of this PR

**Review by leveretka — COMMENTED** [review]
I won't do 'Request changes' to not block you next week.

File: README.md [hunk 7]
```diff
@@ -1,33 +1,48 @@
-Code Quality and Security for Kotlin
-==========
+# Code Quality and Security for Kotlin
```
[the remark this answers is not in the document]
Comment (kaufco) [reply README.md:1]: Fixed. Removed README changes from PR.

File: kotlin-checks-test-sources/src/main/kotlin/checks/ProviderGetOutsideTaskCheckSample.kts [hunk 8]
```diff
@@ -0,0 +1,8 @@
+  println("Hello!") // Compliant
```
[the remark this answers is not in the document]
Comment (kaufco) [reply kotlin-checks-test-sources/src/main/kotlin/checks/ProviderGetOutsideTaskCheckSample.kts:1]: Fixed.

File: kotlin-checks-test-sources/src/main/kotlin/checks/ProviderGetOutsideTaskCheckSample.kts [hunk 9]
```diff
@@ -0,0 +1,8 @@
...
+  pruntln("Hello!") // Noncompliant {{Use "println" instead of "pruntln"}}
+//^^^^^^^
+  println("Hello!") // Compliant
+
+  fun pruntln(value: String) {
+    println(value)
+  }
```
[the remark this answers is not in the document]
Comment (kaufco) [reply kotlin-checks-test-sources/src/main/kotlin/checks/ProviderGetOutsideTaskCheckSample.kts:8]: Fixed.

File: sonar-kotlin-gradle/.gitignore [hunk 10]
```diff
@@ -0,0 +1,42 @@
+.gradle
```
[the remark this answers is not in the document]
Comment (kaufco) [reply sonar-kotlin-gradle/.gitignore:1]: Fixed.

File: sonar-kotlin-plugin/src/main/resources/org/sonar/l10n/kotlin/rules/kotlin/Sonar_way_profile.json (hunk 6 above)
Comment (kaufco) [reply sonar-kotlin-plugin/src/main/resources/org/sonar/l10n/kotlin/rules/kotlin/Sonar_way_profile.json:116]: Fixed.

**Review by johann-beleites-sonarsource — APPROVED** [review]
Let's update the PR title, LGTM!

**Pull request merged**